### PR TITLE
fix: async display cloud infer result

### PR DESCRIPTION
**Pull request by alvindaiyan**
## Description

allow to display the cloud inference result asynchronously 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### CODE REVIEW

File: scripts/main.py [hunk 1]
```diff
@@ -91,14 +103,57 @@ def after_component(self, component, **kwargs):
...
+        if type(component) is gr.HTML and getattr(component, 'elem_id', None) == 'html_info_txt2img' and self.is_txt2img:
+            self.txt2img_html_info = component
+
+        async def _update_result():
+            if self.inference_queue and len(self.inference_queue) > 0:
+                inference_id = self.inference_queue.pop(0)
+                self.latest_result = sagemaker_ui.process_result_by_inference_id(inference_id)
+                return self.latest_result
+
+            return gr.skip(), gr.skip(), gr.skip()
```
Comment (yike5460) [inline scripts/main.py:122]: in which condition?

Comment (yike5460) [reply scripts/main.py:122]: when update but no actual inference execute?

File: scripts/main.py [hunk 2]
```diff
@@ -307,6 +362,7 @@ def encode_no_json(obj):
                                         headers={'x-api-key': api_key})
                 response.raise_for_status()
                 self.current_inference_id = inference_id
+                self.inference_queue.append(inference_id)
```
Comment (yike5460) [inline scripts/main.py:365]: is it possible the queue can grow unlimited and if _update_result() never been called?

**Review by yike5460 — COMMENTED** [review]
make sure the loop & queue are working expected

File: scripts/main.py (hunk 2 above)
Comment (alvindaiyan) [reply scripts/main.py]: fixed, with sized queue

File: scripts/main.py (hunk 1 above)
Comment (alvindaiyan) [reply scripts/main.py:123]: will keep the current image in the page, empty if no image in current. this feature is tested locally

**Review by yike5460 — APPROVED** [review]
lgtm, more test needed still

**Pull request merged**